nautobot/extras/models/groups.py

```diff
@@ -0,0 +1,283 @@
+            except models.ObjectDoesNotExist:
+                dynamicgroupmap_class = None
+
+            self._map = dynamicgroupmap_class
+
+        return self._map
+
+    def get_absolute_url(self):
+        return reverse("extras:dynamicgroup", kwargs={"slug": self.slug})
+
+    def get_group_members_url(self):
+        """Get URL to group members."""
+        base_url = self.map.base_url
+        filter_str = self.map.urlencode(self.filter)
+
+        if filter_str is not None:
+            base_url += f"?{filter_str}"
+
+        return base_url
+
+    def get_filter_fields(self):
+        # Do not present the list of filter options until the object has been created and has a map
+        # class.
+        if not self.map:
+            return {}
+
+        if not self.present_in_database:
+            return {}
+
+        # Add all fields defined in the DynamicGroupMap to the form and populate the default value
+        fields = {}
+        for field_name, field in self.map.fields().items():
+            if field_name in self.filter:
+                field.initial = self.filter[field_name]
+
+            fields[field_name] = field
+
+        return fields
+
+    def set_filter(self, form_data):
+        """
+        Set all desired fields from `form_data` into `filter` dict.
+
+        :param form_data:
+            Dict of filter parameters, generally from a filter form's `cleaned_data`
+        """
+        # Get the authoritative source of filter fields we want to keep.
+        filter_fields = self.get_filter_fields()
+
+        # Populate the filterset from the incoming `form_data`. The filterset's internal form is
+        # used for validation, will be used by us to extract cleaned data for final processing.
+        filterset_class = self.map.filterset_class
+        filterset_class.form_prefix = "filter"
+        filterset = filterset_class(form_data)
+
+        # Use the auto-generated filterset form perform creation of the filter dictionary.
+        filterset_form = filterset.form
+
+        # It's expected that the incoming data has already been cleaned by a form. This `is_valid()`
+        # call is primarily to reduce the fields down to be able to work with the `cleaned_data` from the
+        # filterset form, but will also catch errors in case a user-created dict is provided instead.
+        if not filterset_form.is_valid():
+            raise ValidationError(filterset_form.errors)
+
+        # Perform some type coercions so that they are URL-friendly and reversible, excluding any
+        # empty/null value fields.
+        new_filter = {}
+        for field_name in filter_fields:
+            field = filterset_form.fields[field_name]
+            field_value = filterset_form.cleaned_data[field_name]
+
+            if isinstance(field, forms.ModelMultipleChoiceField):
+                field_to_query = field.to_field_name or "pk"
+                logger.debug("%s - %s", field_value, field_to_query)
```

Comment:
Might be good to have a more verbose log message here, perhaps including `field_name` or at least `self`?